core/templates/dev/head/pages/exploration_player/PlaythroughService.js

```diff
@@ -49,6 +49,20 @@ oppia.factory('PlaythroughService', [
     var cycleIdentifier = {};
     var visitedStates = [];
 
+    /**
+     * A quit action is recorded every time a MaybeQuit event is encountered.
+     * This function extracts the old quit action out of the playthrough
+     * actions list.
+     */
+    var extractOldQuitAction = function() {
+      for (i = 0; i < playthrough.actions.length; i++) {
+        var action = playthrough.actions[i].actionType;
+        if (action === ACTION_TYPE_EXPLORATION_QUIT) {
+          playthrough.actions.splice(i, 1);
```

Comment:
Splicing an array while iterating over it can cause buggy behavior. It would be safer to use `filter` to create an entirely new array:

```javascript
playthrough.actions = playthrough.actions.filter(function(action) {
  return action.actionType !== ACTION_TYPE_EXPLORATION_QUIT;
});
```

Reply:
Makes sense. Done.